File: Algorithms/01-Warmup/08-birthday-cake-candles/08.js
process.stdin.resume();
process.stdin.setEncoding('ascii');

var input_stdin = "";
var input_stdin_array = "";
var input_currentline = 0;

process.stdin.on('data', function (data) {
    input_stdin += data;
});

process.stdin.on('end', function () {
    input_stdin_array = input_stdin.split("\n");
    main();
});

function readLine() {
    return input_stdin_array[input_currentline++];
}

/////////////// ignore above this line ////////////////////

function birthdayCakeCandles(n, ar) {
    var max = 0;
    var count = 0;
    for (var i = 0; i < n; i++) {
        if (ar[i] > max) {
            count = 0;
            max = ar[i];
        }
        if (ar[i] == max) {
            count += 1;
        }
    }
    return count;
}

function main() {
    var n = parseInt(readLine());
    var ar = readLine().split(" ");
    ar = ar.map(Number);
    var result = birthdayCakeCandles(n, ar);
    process.stdout.write("" + result + "\n");

}

File: Algorithms/01-Warmup/08-birthday-cake-candles/08.java
import java.util.Scanner;

class Solution {

    static int birthdayCakeCandles(int n, int[] ar) {
        int max = 0;
        int count = 0;
        for (int i = 0; i < n; i++) {
            if (ar[i] > max) {
                count = 0;
                max = ar[i];
            }
            if (ar[i] == max) {
                count += 1;
            }
        }
        return count;
    }

    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();
        int[] ar = new int[n];
        for(int ar_i = 0; ar_i < n; ar_i++){
            ar[ar_i] = in.nextInt();
        }
        int result = birthdayCakeCandles(n, ar);
        System.out.println(result);
    }
}
